overlord/snapstate/policy/os.go

```diff
@@ -44,7 +44,10 @@ func (p *osPolicy) CanRemove(st *state.State, snapst *snapstate.SnapState, rev s
 		return nil
 	}
 
-	if p.modelBase == "" {
+	// if the base is unset and dev.IsCoreBoot is true, then we know this is a
+	// UC16 system. note that base might be unset on classic models, so we must
+	// check dev.IsCoreBoot as well.
+	if p.modelBase == "" && dev.IsCoreBoot() {
```

Comment:
is core boot true for classic system booting with a kernel snap?

Reply:
yes